broker-core/src/test/java/io/zeebe/broker/workflow/gateway/EventbasedGatewayTest.java

```diff
@@ -215,6 +229,33 @@ public void shouldContinueWhenTimerIsTriggered() {
             .exists());
   }
 
+  @Test
+  public void shouldOnlyExecuteOneBranchWithEqualTimers() {
+    // given
+    testClient.deploy(WORKFLOW_WITH_EQUAL_TIMERS);
+    testClient.createWorkflowInstance(PROCESS_ID);
+    assertThat(RecordingExporter.timerRecords(TimerIntent.CREATED).limit(2).exists()).isTrue();
+    // when
+    brokerRule.getClock().addTime(Duration.ofSeconds(2));
+
+    // then
+    assertThat(
+            RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.GATEWAY_ACTIVATED)
+                .limit(1)
+                .count())
+        .isEqualTo(1);
+
+    assertThat(
+            RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN)
+                    .withElementId("to-end2")
+                    .exists()
+                ^ RecordingExporter.workflowInstanceRecords(
+                        WorkflowInstanceIntent.SEQUENCE_FLOW_TAKEN)
+                    .withElementId("to-end1")
+                    .exists())
+        .isTrue();
```

Comment:
This assertion would wait more than 5 seconds because only one of the events is present. I think it would be better to check that the second timer command is rejected.